src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerConfigTests.cs

```diff
@@ -932,6 +936,81 @@ public void EditorConfigToDiagnostics()
             }, options.Select(o => o.TreeOptions).ToArray());
         }
 
+        [Theory]
+        [InlineData("default")]
+        [InlineData("none")]
+        [InlineData("silent")]
+        [InlineData("refactoring")]
+        [InlineData("suggestion")]
+        [InlineData("warning")]
+        [InlineData("error")]
+        public void EditorConfigToDiagnosticsValidSeverity(string severity)
+        {
+            var expected = severity switch
+            {
+                "default" => ReportDiagnostic.Default,
+                "none" => ReportDiagnostic.Suppress,
+                "silent" => ReportDiagnostic.Hidden,
+                "refactoring" => ReportDiagnostic.Hidden,
+                "suggestion" => ReportDiagnostic.Info,
+                "warning" => ReportDiagnostic.Warn,
+                "error" => ReportDiagnostic.Error,
```

Comment:
would prefer these just be arguments to this test method, so the switch isn't needed.

Reply:
> would prefer these just be arguments to this test method, so the switch isn't needed.

➡️ The current form is intentional. Arguments to the test method are input state and are used to define the unique ID of the test itself. Expected results should not be inputs, and changing an expected outcome should not be part of the test identity.

Reply:
I'd agree with Cyrus here -- we normally put the expected as an input too...

Reply:
> we normally put the expected as an input too

➡️ I try to fix these when they occur in a method I'm working on. Test parameterization should only ever exist on the inputs, not the outputs.